log.py

```diff
@@ -52,33 +53,221 @@ def format(self, record):
+        (internal_log_level, internal_log_format, database_log_level,
+            message_template) = cls._defaults(testing)
+        app_name = cls.DEFAULT_APP_NAME
+
+        if _db and not testing:
+            goal = ExternalIntegration.LOGGING_GOAL
+            internal = ExternalIntegration.lookup(
+                _db, ExternalIntegration.INTERNAL_LOGGING, goal
+            )
+
+            if internal:
+                internal_log_format = (
+                    internal.setting(cls.LOG_FORMAT).value
+                    or internal_log_format
+                )
+                message_template = (
+                    internal.setting(cls.LOG_MESSAGE_TEMPLATE).value
+                    or message_template
+                )
+                internal_log_level = (
+                    ConfigurationSetting.sitewide(_db, Configuration.LOG_LEVEL).value
+                    or internal_log_level
+                )
+                database_log_level = (
+                    ConfigurationSetting.sitewide(_db, Configuration.DATABASE_LOG_LEVEL).value
+                    or database_log_level
+                )
+                app_name = ConfigurationSetting.sitewide(_db, Configuration.LOG_APP_NAME).value or app_name
+
+        handler = logging.StreamHandler()
+        cls.set_formatter(handler, internal_log_format, message_template, app_name)
+
+        return (handler, internal_log_level, database_log_level)
+
+    @classmethod
+    def set_formatter(cls, handler, log_format, message_template, app_name):
+        """Tell the given `handler` to format its log messages in a
+        certain way.
+        """
+        if (log_format == cls.JSON_LOG_FORMAT):
+            formatter = JSONFormatter(app_name)
+        else:
+            formatter = UTF8Formatter(message_template)
+        handler.setFormatter(formatter)
+
+class Loggly(Logger):
+
+    NAME = 'loggly'
+    DEFAULT_LOGGLY_URL = "https://logs-01.loggly.com/inputs/%(token)s/tag/python/"
+
+    USER = 'user'
+    PASSWORD = 'password'
+    URL = 'url'
+
+    SETTINGS = [
+        { "key": USER, "label": _("Username") },
+        { "key": PASSWORD, "label": _("Password") },
+        { "key": URL, "label": _("URL") },
+    ]
+
+    SITEWIDE = True
+
+    @classmethod
+    def from_configuration(cls, _db, testing=False):
+        loggly = None
+        from model import (ExternalIntegration, ConfigurationSetting)
+
+        app_name = cls.DEFAULT_APP_NAME
+        if _db and not testing:
+            goal = ExternalIntegration.LOGGING_GOAL
+            loggly = ExternalIntegration.lookup(
+                _db, ExternalIntegration.LOGGLY, goal
+            )
+            app_name = ConfigurationSetting.sitewide(_db, Configuration.LOG_APP_NAME).value or app_name
+
+        if loggly:
+            loggly = Loggly.loggly_handler(loggly)
+            cls.set_formatter(loggly, cls.LOG_FORMAT, cls.DEFAULT_MESSAGE_TEMPLATE, app_name)
```

Comment:
This class doesn't need any references to LOG_FORMAT or DEFAULT_MESSAGE_TEMPLATE. It's okay for this set_formatter() to have a different signature, because it's only called from from_configuration().